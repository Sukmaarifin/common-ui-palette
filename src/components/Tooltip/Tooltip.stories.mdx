import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { Grid, Button } from '@material-ui/core';
import LinkTo from '@storybook/addon-links/react';

import { MergeOutlinedIcon } from '../../icons/';
import Tooltip from './Tooltip';

<Meta
  component={Tooltip}
  id="Components|Tooltip"
  title="Components|Tooltip"
/>

# Tooltip

`Tooltip` can be used to display additional information on user actions (e.g. hover, focus, etc).
`Tooltip` is a simple wrapper of [MaterialUI Tooltip](https://material-ui.com/components/tooltips/) with custom styling. All of its props are inherited to the `Tooltip`

---

---

---

## Usage

Here are some common usage of `Tooltip`. You can check [MaterialUI Tooltip Doc](https://material-ui.com/components/tooltips/) for more advance usage.

### Basic

You can wrap the children with `Tooltip` which will show the `Tooltip` when users hover, focus, or tap to it.

<Preview>
  <Story name="Basic">
    <Tooltip title="Tooltip can be used to display additional information on user actions (e.g. hover, focus, etc)">
      <Button background="primary">Hover on me! I have a tooltip.</Button>
    </Tooltip>
  </Story>
</Preview>

### Placement

We can set initial `Tooltip` placement. And it will automatically adapt the placement if there is no space available.

<Preview>
  <Story name="Placement">
    <Grid>
      <Tooltip placement="top" title="I should popping up on the top!">
        <Button>Top</Button>
      </Tooltip>
      <Tooltip placement="right" title="I should popping up on the right!">
        <Button>Right</Button>
      </Tooltip>
      <Tooltip
        placement="bottom"
        title="I should popping up on the bottom!"
      >
        <Button>Bottom</Button>
      </Tooltip>
      <Tooltip placement="left" title="I should popping up on the left!">
        <Button>Left</Button>
      </Tooltip>
    </Grid>
  </Story>
</Preview>

### Image

<Preview>
  <Story name="Image">
    <Grid>
      <Tooltip
        placement="top"
        title="I should popping up on the top!"
        img="connexi-logo.png"
      >
        <Button>Tooltip with image</Button>
      </Tooltip>
    </Grid>
  </Story>
</Preview>

## Props

<Props of={Tooltip} />

## Related

- <LinkTo kind="Styleguide|Iconography">Iconography</LinkTo>
