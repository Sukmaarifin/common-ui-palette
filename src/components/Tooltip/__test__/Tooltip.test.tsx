import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Tooltip, { TooltipProps } from '../Tooltip';

describe('Prius tooltip test', () => {
  const render = () => {
    const props: TooltipProps = {
      children: <button>aku</button>,
      placement: 'top',
      title: 'Lorem ipsum',
    };

    return mount(<Tooltip {...props} />);
  };

  it('render successfully', () => {
    const wrapper = render();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('render with img props', () => {
    const preview = 'default-user-img.jpg';
    const wrapper = mount(
      <Tooltip
        children={<button></button>}
        placement={'top'}
        title={'Lorem ipsum'}
        img={preview}
      />
    );
    expect(wrapper.prop('img')).toEqual(preview);
  });
});
