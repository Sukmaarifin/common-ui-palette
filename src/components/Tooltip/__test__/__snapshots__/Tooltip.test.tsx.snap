// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prius tooltip test render successfully 1`] = `
<TooltipComponent
  placement="top"
  title="Lorem ipsum"
>
  <WithStyles(ForwardRef(Tooltip))
    PopperProps={
      Object {
        "disablePortal": false,
        "popperOptions": Object {
          "modifiers": Object {
            "arrow": Object {
              "element": null,
              "enabled": false,
            },
          },
        },
      }
    }
    classes={
      Object {
        "popper": "makeStyles-popper-2",
        "tooltip": "makeStyles-tooltip-3",
        "tooltipPlacementBottom": "makeStyles-tooltipPlacementBottom-7",
        "tooltipPlacementLeft": "makeStyles-tooltipPlacementLeft-4",
        "tooltipPlacementRight": "makeStyles-tooltipPlacementRight-5",
        "tooltipPlacementTop": "makeStyles-tooltipPlacementTop-6",
      }
    }
    disableHoverListener={false}
    placement="top"
    title={
      <React.Fragment>
        <div
          className="prius-tooltip"
        >
          Lorem ipsum
        </div>
        <span
          className="makeStyles-arrow-1"
        />
      </React.Fragment>
    }
  >
    <ForwardRef(Tooltip)
      PopperProps={
        Object {
          "disablePortal": false,
          "popperOptions": Object {
            "modifiers": Object {
              "arrow": Object {
                "element": null,
                "enabled": false,
              },
            },
          },
        }
      }
      classes={
        Object {
          "arrow": "MuiTooltip-arrow",
          "popper": "MuiTooltip-popper makeStyles-popper-2",
          "popperArrow": "MuiTooltip-popperArrow",
          "popperInteractive": "MuiTooltip-popperInteractive",
          "tooltip": "MuiTooltip-tooltip makeStyles-tooltip-3",
          "tooltipArrow": "MuiTooltip-tooltipArrow",
          "tooltipPlacementBottom": "MuiTooltip-tooltipPlacementBottom makeStyles-tooltipPlacementBottom-7",
          "tooltipPlacementLeft": "MuiTooltip-tooltipPlacementLeft makeStyles-tooltipPlacementLeft-4",
          "tooltipPlacementRight": "MuiTooltip-tooltipPlacementRight makeStyles-tooltipPlacementRight-5",
          "tooltipPlacementTop": "MuiTooltip-tooltipPlacementTop makeStyles-tooltipPlacementTop-6",
          "touch": "MuiTooltip-touch",
        }
      }
      disableHoverListener={false}
      placement="top"
      title={
        <React.Fragment>
          <div
            className="prius-tooltip"
          >
            Lorem ipsum
          </div>
          <span
            className="makeStyles-arrow-1"
          />
        </React.Fragment>
      }
    >
      <button
        aria-describedby={null}
        className=""
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title={null}
      >
        aku
      </button>
      <ForwardRef(Popper)
        anchorEl={
          <button
            class=""
          >
            aku
          </button>
        }
        className="MuiTooltip-popper makeStyles-popper-2"
        disablePortal={false}
        id={null}
        open={false}
        placement="top"
        popperOptions={
          Object {
            "modifiers": Object {
              "arrow": Object {
                "element": null,
                "enabled": false,
              },
            },
          }
        }
        transition={true}
      />
    </ForwardRef(Tooltip)>
  </WithStyles(ForwardRef(Tooltip))>
</TooltipComponent>
`;
